napi-inl.h

```diff
@@ -2595,12 +2595,110 @@ inline Error::Error(napi_env env, napi_value value) : ObjectReference(env, nullp
   if (value != nullptr) {
     napi_status status = napi_create_reference(env, value, 1, &_ref);
 
+    // Creates a wrapper object containg the error value (primitive types) and
+    // create a reference to this wrapper
+    if (status != napi_ok) {
+      napi_value wrappedErrorObj;
+      status = napi_create_object(env, &wrappedErrorObj);
+
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_create_object");
+
+      napi_property_descriptor errValDesc = {
+          "errorVal",                     // const char* utf8Name
+          String::From(env, "errorVal"),  //  napi_value name
+          nullptr,                        // Method
+          nullptr,                        // getter
+          nullptr,                        // setter
+          Value::From(env, value),        // napi_value value
+          napi_enumerable,
+          nullptr};
+
+      status = napi_define_properties(env, wrappedErrorObj, 1, &errValDesc);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
+
+      MaybeOrValue<Symbol> result = Symbol::For(env, "isWrapObject");
+      napi_property_descriptor wrapObjFlag = {nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              Value::From(env, value),
+                                              napi_enumerable,
+                                              nullptr};
+
+#ifdef NODE_ADDON_API_ENABLE_MAYBE
+      Symbol uniqueSymb;
+      if (result.IsJust()) {
+        uniqueSymb = result.Unwrap();
+      }
+
+      wrapObjFlag.name = uniqueSymb;
+      status = napi_define_properties(env, wrappedErrorObj, 1, &wrapObjFlag);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
+#else
+
+      wrapObjFlag.name = result;
+      status = napi_define_properties(env, wrappedErrorObj, 1, &wrapObjFlag);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
+
+#endif
+      status = napi_create_reference(env, wrappedErrorObj, 1, &_ref);
+    }
+
     // Avoid infinite recursion in the failure case.
     // Don't try to construct & throw another Error instance.
     NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_create_reference");
   }
 }
 
+inline Object Error::Value() const {
+  if (_ref == nullptr) {
+    return Object(_env, nullptr);
+  }
+
+  napi_value refValue;
+  napi_status status = napi_get_reference_value(_env, _ref, &refValue);
+  NAPI_THROW_IF_FAILED(_env, status, Object());
+
+  napi_valuetype type;
+  status = napi_typeof(_env, refValue, &type);
+  NAPI_THROW_IF_FAILED(_env, status, Object());
+
+  if (type != napi_symbol) {
+    // We are checking if the object is wrapped
+    bool isWrappedObject = false;
+
+    MaybeOrValue<Symbol> result = Symbol::For(_env, "isWrapObject");
```

Comment:
Should this also be guarded by `#ifdef NODE_ADDON_API_ENABLE_MAYBE` since it's `MaybeOrValue<Symbol>`?